modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/helper/PortletPreferencesHelper.java

```diff
@@ -34,31 +35,43 @@ public PortletPreferencesHelper(
 	}
 
 	public boolean getBoolean(String key, boolean defaultValue) {
-		Optional<Boolean> valueOptional = getBoolean(key);
-
-		return valueOptional.orElse(defaultValue);
+		return GetterUtil.getBoolean(_getValue(key), defaultValue);
 	}
 
 	public int getInteger(String key, int defaultValue) {
-		Optional<Integer> valueOptional = getInteger(key);
-
-		return valueOptional.orElse(defaultValue);
+		return GetterUtil.getInteger(_getValue(key), defaultValue);
 	}
 
-	public Optional<String> getString(String key) {
+	public String getString(String key) {
 		return _getValue(key);
 	}
 
 	public String getString(String key, String defaultValue) {
-		Optional<String> valueOptional = getString(key);
+		String value = getString(key);
 
-		return valueOptional.orElse(defaultValue);
+		if (value == null) {
+			return defaultValue;
+		}
+
+		return value;
 	}
 
-	private Optional<String> _getValue(String key) {
-		return _portletPreferencesOptional.flatMap(
-			portletPreferences -> SearchStringUtil.maybe(
-				portletPreferences.getValue(key, StringPool.BLANK)));
+	private String _getValue(String key) {
+		if (!_portletPreferencesOptional.isPresent()) {
```

Comment:
@julschong Please make this field non-optional, and just do `field = optional.orElse(null)` in the constructor. Then when you modify the api later you can simply touch the constructor.